Source/CarthageKit/Xcode.swift

```diff
@@ -971,31 +984,52 @@ public func buildInDirectory(directoryURL: NSURL, withConfiguration configuratio
 				return false
 			}
 		}
-		|> take(1)
-		|> flatMap(.Merge) { (project: ProjectLocator) -> SignalProducer<String, CarthageError> in
+		|> flatMap(.Concat) { (project: ProjectLocator) -> SignalProducer<[SharedScheme], CarthageError> in
 			return schemesInProject(project)
+				|> flatMap(.Merge) { scheme -> SignalProducer<SharedScheme, CarthageError> in
+					let buildArguments = BuildArguments(project: project, scheme: scheme, configuration: configuration)
+
+					return shouldBuildScheme(buildArguments, platform)
+						|> filter { $0 }
+						|> map { _ in SharedScheme(schemeName: scheme, project: project) }
+				}
+				|> collect
+				|> flatMap(.Concat) { schemes in
+					if schemes.isEmpty {
+						return .empty
+					} else {
+						return SignalProducer(value: schemes)
+					}
+				}
```

Comment:
I think you could simplify this a bit by replacing the `flatMap` with a `filter`:

```swift
|> filter { !$0.isEmpty }
```
```